dulwich/index.py

```diff
@@ -629,17 +629,28 @@ def _has_directory_changed(tree_path, entry):
     return False
 
 
-def get_unstaged_changes(index, root_path, filter_blob_callback=None):
+def get_unstaged_changes(index, repo, filter_blob_callback=None):
```

Comment:
It seems like a layer violation to pass in a repo here. Do you really need the whole repo, or could you e.g. just pass in a different filter_blob_callback?